src/simulator/quickbeam/quickbeam.F90

```diff
@@ -372,19 +379,29 @@ subroutine cloudsat_precipOccurence(Npoints, Ncolumns, llm, Nhydro, Ze_out, Ze_n
     integer,dimension(Npoints,Ncolumns) :: &
          cloudsat_pflag,      & ! Subcolumn precipitation flag
          cloudsat_precip_pia    ! Subcolumn path integrated attenutation.
-    integer :: pr,i,k,m,j
+    integer,dimension(Npoints) :: &                                           !PREC_BUG
+         cloudsat_preclvl_index ! Altitude index for precip flags calculation !PREC_BUG
+                                ! in 40-level grid (one layer above surfelev) !PREC_BUG     
+!    integer :: pr,i,k,m,j                                                    !PREC_BUG
+     integer :: pr,i,k,m,j,cloudsat_preclvl                                   !PREC_BUG
     real(wp) :: Zmax
     
     ! Initialize 
     cloudsat_pflag(:,:)        = pClass_default
     cloudsat_precip_pia(:,:)   = 0._wp
     cloudsat_precip_cover(:,:) = 0._wp
     cloudsat_pia(:)            = 0._wp
+    cloudsat_preclvl_index(:)  = 0._wp !PREC_BUG
+
+!!! Computing altitude index for precip flags calculation  !PREC_BUG
+cloudsat_preclvl_index(:) = 39 - floor( surfelev(:)/480. ) !PREC_BUG
```

Comment:
As mentioned by @RobertPincus , we should change these magic numbers. In fact, they are not constants because there is some flexibility in the selection of number of levels and depth of the layers, which are calculated in COSP_INIT. They should be taken from the actual definition of the vertical grid. If the diagnostics should not be calculated when csat_vgrid=.false., then they need to be protected.